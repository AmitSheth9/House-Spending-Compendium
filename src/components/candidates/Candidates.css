.pol-container {
    border: 5px solid brown;
    margin-bottom: 10px;
    width: 60%;
    margin-left: 10px;
    padding: 20px;
    margin-top: 10px;
}
p {
    color: red;
    font-weight: 600;
    font-size: 20px
}
.name {
    font-size: 35px;
}
.top-lines {
    font-size: 25px;
    font-weight: 500;
}
body {
    background: lightgoldenrodyellow;
}