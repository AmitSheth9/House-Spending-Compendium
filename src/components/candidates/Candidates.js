import React from 'react'
import './Candidates.css'

export default function Candidates({candidateArr}) {
    return (
        <div>
            {candidateArr.map((candidate => {
                return <div className='pol-container' key={candidate.candidate_id}>
                    <p className='name'>{candidate.name}</p>
                    <p className='top-lines'>Receipts: {candidate.receipts}</p>
                    <p className='top-lines'>Disbursements: {candidate.disbursements}</p>
                    <br/>
                    <p>Party: {candidate.party_full}</p>
                    <p>Cycle: {candidate.cycle}</p>
                    <p>State: {candidate.state}</p>
                    <p>Cash on Hand: {candidate.cash_on_hand_end_period}</p>
                    </div>
            }))}
        </div>
    )
}
