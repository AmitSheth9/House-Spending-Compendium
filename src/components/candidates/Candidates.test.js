import {screen, render} from '@testing-library/react'
import Candidates from './Candidates'

it('should render our Candidates component', () => {
    const {container} = render(<Candidates candidateArr={[{name:'john', receipts: 30000, disbursements: 4000, party_full: 'pickle party', cycle: 2020, state: 'ca', cash_on_hand_end_period: 3000}]}/>)

expect(container).toMatchSnapshot();
});