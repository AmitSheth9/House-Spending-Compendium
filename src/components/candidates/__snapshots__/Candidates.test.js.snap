// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Candidates component 1`] = `
<div>
  <div>
    <div
      class="pol-container"
    >
      <p
        class="name"
      >
        john
      </p>
      <p
        class="top-lines"
      >
        Receipts: 
        30000
      </p>
      <p
        class="top-lines"
      >
        Disbursements: 
        4000
      </p>
      <br />
      <p>
        Party: 
        pickle party
      </p>
      <p>
        Cycle: 
        2020
      </p>
      <p>
        State: 
        ca
      </p>
      <p>
        Cash on Hand: 
        3000
      </p>
    </div>
  </div>
</div>
`;
